dart-lang/src/main/java/fr/insideapp/sonarqube/dart/lang/issues/dartanalyzer/DartAnalyzerSensor.java

```diff
@@ -46,243 +46,293 @@
+                    .toString();
+
+            LOGGER.debug("Current file full path: {}", fullPath);
+
+            filesWithAbsolutePath.add(fullPath);
+        });
+
+        return filesWithAbsolutePath;
+    }
+
+    private FilePredicate getFilesFilter(FileSystem fileSystem) {
+        FilePredicate mainFilePredicate = fileSystem.predicates().and(
+                fileSystem.predicates().hasType(InputFile.Type.MAIN), fileSystem.predicates().hasLanguage(Dart.KEY));
+        return mainFilePredicate;
+    }
+
+    private FileSystem getFileSystem(SensorContext sensorContext) {
+        FileSystem fileSystem = sensorContext.fileSystem();
+        return fileSystem;
+    }
+
+    private void recordIssues(SensorContext sensorContext, List<DartAnalyzerReportIssue> issues) {
+        // Record issues
+        issues.forEach(i -> {
+            File file = new File(sensorContext.fileSystem().baseDir(), i.getFilePath());
+            LOGGER.debug("Inside issue forEach, file absolute path: {}", file.getAbsolutePath());
+
+            FilePredicate fp = sensorContext.fileSystem().predicates().hasAbsolutePath(file.getAbsolutePath());
+            if (!sensorContext.fileSystem().hasFiles(fp)) {
+                LOGGER.warn("File not included in SonarQube {}", file.getAbsoluteFile());
+            } else {
+                InputFile inputFile = sensorContext.fileSystem().inputFile(fp);
+                NewIssueLocation nil = new DefaultIssueLocation().on(inputFile)
+                        .at(inputFile.selectLine(i.getLineNumber())).message(i.getMessage());
+                sensorContext.newIssue().forRule(RuleKey.of(DartAnalyzerRulesDefinition.REPOSITORY_KEY, i.getRuleId()))
+                        .at(nil).save();
+            }
+        });
+    }
+
+    private void verifyIfDartAnalyzerExists() {
+        LOGGER.debug("Verify dart analyser...");
+        new ProcBuilder(ANALYZER_DART_COMMAND).withArg("-h").run();
+        LOGGER.debug("Verify dart analyser done");
+    }
+
+    private boolean existsAnalysisOptionsFile(SensorContext sensorContext) {
+        return new File(sensorContext.fileSystem().baseDir(), ANALYSIS_OPTIONS_FILENAME).exists();
+    }
+
+    private void saveCurrentAnalysisOptionsFile(SensorContext sensorContext) {
+        File analysisOptionsFile = new File(sensorContext.fileSystem().baseDir(), ANALYSIS_OPTIONS_FILENAME);
+        if (analysisOptionsFile.exists()) {
+            analysisOptionsFile
+                    .renameTo(new File(sensorContext.fileSystem().baseDir(), ANALYSIS_OPTIONS_FILENAME + ".sonar"));
+            LOGGER.info("Backup of original analysis_options.yaml file to {}", ANALYSIS_OPTIONS_FILENAME + ".sonar");
+        }
+    }
+
+    private void createAnalysisOptionsFile(SensorContext sensorContext) throws IOException {
+        File analysisOptionsFile = new File(sensorContext.fileSystem().baseDir(), ANALYSIS_OPTIONS_FILENAME);
+        URL inputUrl = getClass().getResource(ANALYSIS_OPTIONS_FILE);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [UI: Usage of GetResource may be unsafe if class is extended (UI_INHERITANCE_UNSAFE_GETRESOURCE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#ui-usage-of-getresource-may-be-unsafe-if-class-is-extended-ui-inheritance-unsafe-getresource).
Calling this.getClass().getResource(...) could give results other than expected if this class is extended by a class in another package.

Reply:
@kieraCurtis Would be `getClass.getClassLoader().getResource(...)` a better option?

Reply:
What about `DartAnalyzerSensor.class.getResource(...)`?

Reply:
@jrehwaldt Good point, probably it's the right choice here

Reply:
```suggestion
        URL inputUrl = DartAnalyzerSensor.class.getResource(ANALYSIS_OPTIONS_FILE);
```
@mreichelt Maybe you can give it a try when you have time

Reply:
It's good to have multiple eyes on this, thanks @kieraCurtis, @matpag and @jrehwaldt. I added the change. 👍

To be fair, I didn't add that line of code in the first place - it was all there all along. You can hide the whitespace changes to see what I actually changed like so:
https://github.com/insideapp-oss/sonar-flutter/pull/47/files?diff=unified&w=1

@zippy1978 and @victorgilc: it seems you are pretty busy with other things - I hope that you are well and that you didn't give up on this project.